### ISSUE TITLE
Make WebXR an optional feature

### ISSUE THREAD

**Issue by frc4533-lincoln** [servo/servo#34195]
**Describe the new feature:**
Add a `webxr` feature on all crates that depend on `webxr` and/or `webxr-api`.
It could be enabled by default. 

WebXR is rarely used, but is a hard requirement for the `canvas` crate.
Additionally `webxr-api` is required by:
 - `canvas_traits`
 - `constellation`
 - `embedder_traits`
 - `script_traits`
 - and all crates that depend on them...

**Comment by msub2** [comment]
There's an existing PR for this at #33526, I'm not sure if it's stalled out because they weren't sure how to add tests for it or if they just haven't gotten back to it.

**Comment by sagudev** [comment]
it waits for some script refactors that would modularize script and allow more separate script moduls (such as webgpu and webxr to be in separate script crates).

**Comment by jdm** [comment]
I don't think it makes sense to wait on the script refactoring. I think that's also going to be a compex project.

**Comment by frc4533-lincoln** [comment]
@msub2 Sorry; I should've looked a little harder.
@sagudev Thanks for letting me know.

Happy to see all the great work going on with Servo.

**Comment by sagudev** [comment]
Duplicate of https://github.com/servo/servo/issues/33464
